modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogTableTest.java

```diff
@@ -0,0 +1,1168 @@
+
+        // Add column.
+        assertThat(manager.execute(addColumnParams(TABLE_NAME, columnParams(NEW_COLUMN_NAME, INT32))), willCompleteSuccessfully());
+        verify(eventListener).notify(any(AddColumnEventParameters.class));
+
+        // Drop column.
+        assertThat(manager.execute(dropColumnParams(TABLE_NAME, NEW_COLUMN_NAME)), willCompleteSuccessfully());
+        verify(eventListener).notify(any(DropColumnEventParameters.class));
+
+        verifyNoMoreInteractions(eventListener);
+    }
+
+    @Test
+    public void testGetCatalogEntityInCatalogEvent() {
+        var fireEventFuture = new CompletableFuture<Void>();
+
+        manager.listen(CatalogEvent.TABLE_CREATE, fromConsumer(fireEventFuture, parameters -> {
+            assertNotNull(manager.schema(parameters.catalogVersion()));
+        }));
+
+        assertThat(manager.execute(simpleTable(TABLE_NAME)), willCompleteSuccessfully());
+        assertThat(fireEventFuture, willCompleteSuccessfully());
+    }
+
+    @Test
+    public void testGetTableByIdAndCatalogVersion() {
+        int tableCreationVersion = await(manager.execute(simpleTable(TABLE_NAME)));
+
+        CatalogTableDescriptor table = manager.table(TABLE_NAME, clock.nowLong());
+
+        assertNull(manager.table(table.id(), tableCreationVersion - 1));
+        assertSame(table, manager.table(table.id(), tableCreationVersion));
+    }
+
+    @Test
+    public void alterColumnIncrementsTableVersion() {
+        createSomeTable(TABLE_NAME);
+
+        CompletableFuture<?> future = manager.execute(
+                AlterTableAlterColumnCommand.builder()
+                        .schemaName(SCHEMA_NAME)
+                        .tableName(TABLE_NAME)
+                        .columnName("val1")
+                        .type(INT64)
+                        .build()
+        );
+        assertThat(future, willCompleteSuccessfully());
+
+        CatalogTableDescriptor table = manager.table(TABLE_NAME, Long.MAX_VALUE);
+
+        assertThat(table.tableVersion(), is(2));
+    }
+
+    /**
+     * Checks for possible changes to the default value of a column descriptor.
+     *
+     * <p>Set/drop default value allowed for any column.
+     */
+    @Test
+    public void testAlterColumnDefault() {
+        assertThat(manager.execute(simpleTable(TABLE_NAME)), willCompleteSuccessfully());
+
+        int schemaVer = manager.latestCatalogVersion();
+        assertNotNull(manager.schema(schemaVer));
+        assertNull(manager.schema(schemaVer + 1));
+
+        // NULL-> NULL : No-op.
+        assertThat(changeColumn(TABLE_NAME, "VAL", null, null, () -> constant(null)),
+                willCompleteSuccessfully());
+        assertNull(manager.schema(schemaVer + 1));
+
+        // NULL -> 1 : Ok.
+        assertThat(changeColumn(TABLE_NAME, "VAL", null, null, () -> constant(1)),
+                willCompleteSuccessfully());
+        assertNotNull(manager.schema(++schemaVer));
+
+        // 1 -> 1 : No-op.
```

Comment:
Should we check a functional default as well?

Reply:
Already covered under `AlterTableAlterColumnCommandValidationTest::cannotAddColumnWithFunctionalDefault`